src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,299 @@
+		$enqueued = array();
+		foreach ( self::$registered as $module_identifier => $module ) {
+			if ( true === $module['enqueued'] ) {
+				$enqueued[ $module_identifier ] = $module;
+			}
+		}
+		return $enqueued;
+	}
+
+	/**
+	 * Retrieves all the dependencies for the given module identifiers, filtered
+	 * by types.
+	 *
+	 * It will consolidate an array containing a set of unique dependencies based
+	 * on the requested types: 'static', 'dynamic', or both. This method is
+	 * recursive and also retrieves dependencies of the dependencies.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $module_identifiers The identifiers of the modules for which to gather dependencies.
+	 * @param array $types              Optional. Types of dependencies to retrieve: 'static', 'dynamic', or both. Default is both.
+	 * @return array List of dependencies, keyed by module identifier.
+	 */
+	private static function get_dependencies( $module_identifiers, $types = array( 'static', 'dynamic' ) ) {
+		return array_reduce(
+			$module_identifiers,
+			function ( $dependency_modules, $module_identifier ) use ( $types ) {
+				$dependencies = array();
+				foreach ( self::$registered[ $module_identifier ]['dependencies'] as $dependency ) {
+					if (
+						in_array( $dependency['type'], $types, true ) &&
+						isset( self::$registered[ $dependency['id'] ] ) &&
+						! isset( $dependency_modules[ $dependency['id'] ] )
+					) {
+						$dependencies[ $dependency['id'] ] = self::$registered[ $dependency['id'] ];
+					}
+				}
+				return array_merge( $dependency_modules, $dependencies, self::get_dependencies( array_keys( $dependencies ), $types ) );
+			},
+			array()
+		);
+	}
+}
+
+/**
+ * Registers the module if no module with that module identifier has already
+ * been registered.
+ *
+ * @since 6.5.0
+ *
+ * @param string            $module_identifier The identifier of the module. Should be unique. It will be used in the final import map.
+ * @param string            $src               Full URL of the module, or path of the script relative to the WordPress root directory.
+ * @param array             $dependencies      Optional. An array of module identifiers of the dependencies of this module. The dependencies can be strings or arrays. If they are arrays, they need an `id` key with the module identifier, and can contain a `type` key with either `static` or `dynamic`. By default, dependencies that don't contain a type are considered static.
+ * @param string|false|null $version           Optional. String specifying module version number. Defaults to false. It is added to the URL as a query string for cache busting purposes. If SCRIPT_DEBUG is true, the version is the current timestamp. If $version is set to false, the version number is the currently installed WordPress version. If $version is set to null, no version is added.
+ */
+function wp_register_module( $module_identifier, $src, $dependencies = array(), $version = false ) {
```

Comment:
This file must only contain the `WP_Modules` class. Any other functions etc. should be in its own file.

Reply:
Moved to their own file in https://github.com/WordPress/wordpress-develop/pull/5818/commits/156eb588ec82771974f3722729b6c4b4685561b3.